File: stockaqui/src/app/shared/model/produto.ts
export class Produto {
    id?: string;
    nome = '';
    valor: number;
    descricao = '';
    fk_estoque = '';

    constructor(id?: string, produto: Produto = {nome: '', valor: 0, descricao: '', fk_estoque: ''}) {
        this.id = id;
        this.nome = produto.nome;
        this.valor = produto.valor;
        this.descricao = produto.descricao;
        this.fk_estoque = produto.fk_estoque;
    }
}

File: api/src/main/java/stockaqui/api/modelo/Produto.java
package stockaqui.api.modelo;

import jakarta.persistence.*;
import lombok.*;
import stockaqui.api.produto.DadosCadastroProduto;

@Table(name = "produtos")
@Entity(name = "Produto")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Produto {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;
    private double valor;
    private String descricao;

    @ManyToOne
    private Estoque estoque;

    public Produto(DadosCadastroProduto dados, Estoque estoque) {
        this.nome = dados.nome();
        this.valor = dados.valor();
        this.descricao = dados.descricao();
        this.estoque = estoque;
    }
}
